log4j-api/src/main/java/org/apache/logging/log4j/spi/ThreadContextMapFactory.java

```diff
@@ -88,7 +89,7 @@ public static ThreadContextMap createThreadContextMap() {
             try {
                 final Class<?> clazz = cl.loadClass(ThreadContextMapName);
                 if (ThreadContextMap.class.isAssignableFrom(clazz)) {
-                    result = (ThreadContextMap) clazz.newInstance();
+                    result = (ThreadContextMap) clazz.getDeclaredConstructor().newInstance();
```

Comment:
Why is this not a `LoaderUtil.newInstanceOf(clazz)`?

Reply:
`LoaderUtil.newInstanceOf` caused problem with internal (package-private) implementations. I added a switch-case for those.